### PR TITLE
Migrate kubernetes-incubator/external-storage/nfs

### PR DESCRIPTION
**Pull request by kmova**
This PR covers the migration of code from https://github.com/kubernetes-incubator/external-storage/tree/master/nfs to this repo. 

The tasks include. 
- [x] import nfs from external-storage repo (git filter-branch --subdirectory-filter nfs -- --all)
- [x] update README.md (3efa64d7e33cc5cf711f1086a8cc630e731a1dae)
- [x] update go import path (eb12c1db9298878cef5b5dad3d0290997fe647e1)
- [x] update with image name and registry location (a1c4fd8d7da4e6da012ecbf45d2a19fbf450d9ad)
- [x] update vendor (0140304ce7f3cbe220c90484f5e235ab1ed46dd2)

With the changes made in this PR, the following build commands work. 
- `make container` - generates `quay.io/external_storage/nfs-ganesha-server-and-provisioner:latest`
- `make clean`

Tasks to be done in future PRs: 
- Setup the new quay registry location
- Refactor E2e
- Setup build automation (PROW) to push containers to registry
- Update the docs/deployment.md to reflect new/updated testing scenarios.

### CODE REVIEW

File: deploy/kubernetes/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    type: Recreate 
+  template:
+    metadata:
+      labels:
+        app: nfs-provisioner
+    spec:
+      serviceAccount: nfs-provisioner
+      containers:
+        - name: nfs-provisioner
+          image: quay.io/kubernetes_incubator/nfs-provisioner:latest
```
Comment (ashishranjan738) [inline deploy/kubernetes/deployment.yaml]: this needs to be changed right ?

File: deploy/kubernetes/pod.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+---
+kind: Pod
+apiVersion: v1
+metadata:
+  name: nfs-provisioner
+spec:
+  serviceAccount: nfs-provisioner
+  containers:
+    - name: nfs-provisioner
+      image: quay.io/kubernetes_incubator/nfs-provisioner:latest
```
Comment (ashishranjan738) [inline deploy/kubernetes/pod.yaml]: same here

File: deploy/kubernetes/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  template:
+    metadata:
+      labels:
+        app: nfs-provisioner
+    spec:
+      serviceAccount: nfs-provisioner
+      terminationGracePeriodSeconds: 10
+      containers:
+        - name: nfs-provisioner
+          image: quay.io/kubernetes_incubator/nfs-provisioner:latest
```
Comment (ashishranjan738) [inline deploy/kubernetes/statefulset.yaml]: same here

**Comment by kmova** [comment]
@jsafrane @msau42 @wongma7 - this is ready for review. 

Also, @msau42 - could you help with removing the check of invalid commit similar to https://github.com/kubernetes-sigs/gluster-block-external-provisioner/pull/1#issuecomment-635445925

**Comment by wongma7** [comment]
IMO it's not necessary to continue using the quay external_storage org. Can move to docker hub, not use an org at all, or whatever makes sense.

**Comment by msau42** [comment]
Now we have community infra supported by wg-k8s-infra. I would reach out to them on how to get started. You can also take a look at the kubernetes-csi repos for examples on how to configure the builds. I think we can do that as a second step after getting the initial commits merged.

**Comment by msau42** [comment]
I've removed the invalid commit message label. I'm not sure how to handle the invalid CLA from the imported commits. @dims do you know?

**Comment by kmova** [comment]
Took the help of linuxfoundation support and found that the issue is with the following older commits, where the github profile doesn't exist now. 
- 69c239b662f4ca15019b2df26766f0931f33a4d6
- 99301c73eaf8c9f3e349d2d82f5c286f7b70377a

**Comment by msau42** [comment]
Yeah, in the past we have force merged them but I'm not sure of the actual mechanics of how to do that. Is it just having an administrator hit the "Merge pull request" button?

**Comment by dims** [comment]
@kmova please open a migration issue in https://github.com/kubernetes/org/issues/new/choose and ping the folks on the github admin team https://github.com/kubernetes/community/tree/master/github-management#github-administration-team and they will be able get the changes in.

**Comment by kmova** [comment]
This PR had an issue with CLA which got resolved and merged via https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/pull/3

**Pull request closed without merging**